util/codec/codec.go

```diff
@@ -352,6 +352,240 @@ func encodeHashChunkRowIdx(sc *stmtctx.StatementContext, row chunk.Row, tp *type
 	return
 }
 
+func writeColumn(w io.Writer, flag, data []byte) (err error) {
+	_, err = w.Write(flag)
+	if err != nil {
+		return
+	}
+	_, err = w.Write(data)
+	return
+}
+
+// HashChunkColumns writes the encoded value of each row's column, which of index `colIdx`, to w.
+// `nullBitmap` used to save the rows whose column of index `colIdx` is `Null`.
+func HashChunkColumns(sc *stmtctx.StatementContext, w []io.Writer, chk *chunk.Chunk, tp *types.FieldType, colIdx int, buf, nullBitmap []byte) (err error) {
+	var (
+		b    []byte
+		rows int
+		row  chunk.Row
+	)
+	rows = chk.NumRows()
```

Comment:
We can use `rows := chk.NumRows()` to initialize the value.

Reply:
have done